### ISSUE TITLE
Add package overrides to targeting packs

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/core-setup#6507]
We should add a text file to the targeting packs that specifies which packages (and versions) should be overridden.  This is used to help optimize conflict resolution.  This information was previously provided in .targets files in the corresponding NuGet packages or in the .NET SDK itself.  (https://github.com/dotnet/sdk/issues/2937 has more details).

The information should go in a `data/PackageOverrides.txt` file with each package and version on a separate line, separated by a pipe.  For example (from ASP.NET Core):

```
Microsoft.Extensions.Caching.Abstractions|3.0.0-preview6.19262.1
Microsoft.Extensions.Caching.Memory|3.0.0-preview6.19262.1
Microsoft.Extensions.Configuration.Abstractions|3.0.0-preview6.19262.1
```

@ericstj can you help us figure out what the overridden packages should be for Microsoft.NETCore.App, Microsoft.WindowsDesktop.App, and NETStandard.Library?  We could just use the lists we used for previous versions, although I would guess there are some new NuGet packages that are also bundled in .NET Core 3, and we might want to override the WindowsCompat packages.

**Comment by ericstj** [comment]
/cc @wtgodbe @eerhardt 

I believe the purpose of this feature is only to include the dead-ended packages which have been absorbed into the framework packages.  For .NETStandard2.1 this would be the shims which were added in 2.1.  NETCoreApp should be the same.  

For the rest I believe we wish to continue letting conflict resolution of file versions decide which versions to include since that is all mechanical and easier for people to reason about.

**Comment by dagood** [comment]
It *looks* like this is a duplicate of https://github.com/dotnet/core-setup/issues/3607, albeit with a different spec. I found the background info there pretty useful, so please correct me if I'm wrong. 😄

> For .NETStandard2.1 this would be the shims which were added in 2.1. NETCoreApp should be the same.

Is it right to interpret this as basically copying the lists in [Microsoft.NET.DefaultPackageConflictOverrides.targets](https://github.com/dotnet/sdk/blob/ec2f3741092f164daeb023e7caa4364c810461e2/src/Tasks/Common/targets/Microsoft.NET.DefaultPackageConflictOverrides.targets) into Core-Setup?

> For the rest I believe we wish to continue letting conflict resolution of file versions decide which versions to include since that is all mechanical (and thus generateable / toolable in the shared framework tooling) and easier for people to reason about.

Does "and thus generateable / toolable in the shared framework tooling" refer to generating the list of overrides when creating the targeting packs? It seems like the wish to continue with file version conflict resolution doesn't match this, if I'm understanding it right....

**Comment by dsplaisted** [comment]
We do want this data file, but it is OK if it is a static list of dead-end packages, as opposed to the including the latest version of every active package that is included in the shared framework.

**Comment by dagood** [comment]
Merged into 3.0 with https://github.com/dotnet/core-setup/pull/7374.
